e2e-tests/src/main/java/dev/streamx/cli/test/tools/terminal/process/StreamDataCollectorThread.java

```diff
@@ -0,0 +1,60 @@
+package dev.streamx.cli.test.tools.terminal.process;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.InterruptedIOException;
+import java.util.concurrent.CopyOnWriteArrayList;
+import org.jboss.logging.Logger;
+
+class StreamDataCollectorThread extends Thread {
+
+  private final Logger logger = Logger.getLogger(StreamDataCollectorThread.class);
+  private final CopyOnWriteArrayList<String> ongoingDataCollection;
```

Comment:
I suppose we could use StringBuffer - it could be more memory effective. 

However it's good enough as output should not be so big...

But it's quite good topic for blog post ;)

Reply:
As we agreed on priv there is a problem with caching stringBuffer by some thread and results are not deterministic. Thus we stay with CopyOnWriteArrayList